modules/node/src/admin/admin.service.ts

```diff
@@ -177,4 +179,60 @@ export class AdminService {
 
     return toMerge;
   }
+
+  async fixProxyFactoryAddresses(): Promise<void> {
+    const channels = await this.channelService.findAll();
+
+    for (const channel of channels) {
+      const { data: stateChannel } = await this.cfCoreService.getStateChannel(
+        channel.multisigAddress,
+      );
+      console.log("stateChannel: ", stateChannel);
+      const contractAddresses = await this.configService.getContractAddresses();
+
+      const expectedMultisigAddress = await getCreate2MultisigAddress(
+        stateChannel.userNeuteredExtendedKeys,
+        stateChannel.proxyFactoryAddress || contractAddresses.ProxyFactory,
+        contractAddresses.MinimumViableMultisig,
+        this.configService.getEthProvider(),
+      );
+
+      if (expectedMultisigAddress === stateChannel.multisigAddress) {
+        if (!stateChannel.proxyFactoryAddress) {
+          logger.log(`Multisig address is as expected, adding correct proxyFactory address`);
+          // save new state channel
+        }
+      } else {
+        // try old proxy factory addresses to get expected address
+      }
+    }
+  }
+
+  async getIncorrectProxyFactoryAddresses(): Promise<string[]> {
+    const channels = await this.channelService.findAll();
+
+    const incorrectChannels = [];
+    for (const channel of channels) {
+      const { data: stateChannel } = await this.cfCoreService.getStateChannel(
+        channel.multisigAddress,
+      );
+      console.log("stateChannel: ", stateChannel);
+      const contractAddresses = await this.configService.getContractAddresses();
```

Comment:
there have been a couple proxy factories so far, we should be fetching all of them

even worse, we were using the `bytecode` in the deps over the contract method iirc, so we will have to make sure the all the used `bytecode` lines up with what's onchain

Reply:
I thought we changed this to be using the contract method. `bytecode` is toxic waste. Where are we storing all the historical proxy factory addresses?